Formula/openssl11.rb

```diff
@@ -0,0 +1,176 @@
+class Openssl11 < Formula
+  desc "Cryptography and SSL/TLS Toolkit"
+  homepage "https://openssl.org/"
+  url "https://www.openssl.org/source/openssl-1.1.0.tar.gz"
+  mirror "https://www.mirrorservice.org/sites/ftp.openssl.org/source/openssl-1.1.0.tar.gz"
+  sha256 "f5c69ff9ac1472c80b868efc1c1c0d8dcfc746d29ebe563de2365dd56dbd8c82"
+
+  keg_only :provided_by_osx,
+    "Apple has deprecated use of OpenSSL in favor of its own TLS and crypto libraries"
+
+  option :universal
+  option "without-test", "Skip build-time tests (not recommended)"
+
+  # Only needs 5.10 to run, but needs >5.13.4 to run the testsuite.
+  # https://github.com/openssl/openssl/blob/4b16fa791d3ad8/README.PERL
+  # The MacOS ML tag is same hack as the way we handle most :python deps.
+  if build.with? "test"
+    depends_on :perl => "5.14" if MacOS.version <= :mountain_lion
+  else
+    depends_on :perl => "5.10"
+  end
+
+  # This isn't technically needed where Clang exists, but is a small
+  # dep and is hard to reliably express non-Clang-only in dependency format.
+  depends_on "makedepend" => :build
```

Comment:
https://github.com/openssl/openssl/commit/30752dd7df7c699b380043cb73e384dd84cab539
https://github.com/openssl/openssl/commit/09aa263a143b006650a0f879dc2348609587b806

Apparently Clang (since always) and some GCC versions support dependency generation natively without having to lean on `makedepend` to do that magic.